### DISCUSSION TITLE
Know the actual size of the storage space used per user

### DISCUSSION THREAD

**Discussion by ghidalg0** [supabase/supabase#18721] [Questions]
Hi all ☀️
I'm currently working on an app where our users will be able to manage their documents. Thus, we would like to know the space currently used per user. Our current architecture, is that we create 1 bucket per user, and then organize their buckets in folders + subfolders. 
The only way I found to compute the size of the bucket, is by retrieving each files one by one and summing their sizes. 
Is there a helper / more straightforward way to do so ?
Best,
Guilhem

**Comment by GaryAustin1** [comment]
There is not a helper.  But the storage.objects table can be accessed via rpc or even changing the schema the client uses to storage instead of public.  You would need to query the metadata column with the owner id as a filter.  The column is json so you would need to either have postgres function sum it or post process the json array you get summing that field.  It looks like `{"size":9359,"mimetype":"image/avif","cacheControl":"max-age=3600"}`

**Reply by ghidalg0** [reply]
Thanks for your anwser ! will try it that way then 👌
